Add note about peephole optimizer in docs (#13474)
ethereum/solidity: docs/internals/optimizer.rst

optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
for a stand-alone Yul mode.

.. note::

>     The PeepholeOptimizer is always enabled by default and can only be turned off via the

Reviewer: Would it make sense to mention what the peephole optimizer is, since I don't think we have it documented anywhere?

Author: From what I gather, `Peephole` refers to a common optimization method, so I guess that's why it does not have its own doc section. Maybe we could add an external link ([wikipedia article](https://en.wikipedia.org/wiki/Peephole_optimization)) as a reference to the technique itself.

Author: ```suggestion
    The `PeepholeOptimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ is always enabled by default and can only be turned off via the
```

>     The `PeepholeOptimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ and the inliner are always

Reviewer 2: Are we referring to optimizer steps or just using generic names? I'd make that consistent one way or another.
```suggestion
    The `peephole optimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ and the inliner are always
```

Author: The other occurrences of the word are as you suggested, so I am changing it.

>     enabled by default and can only be turned off via the :ref:`JSON-input-output interface <compiler-api>`.

Reviewer 2: Peephole optimizer is not the only such step. The other one is the inliner:
>        // The inliner is always on if no details are given,
>        // use details to switch it off.
>        "inliner": true,

Can you mention it too?

Author: Done.

>     enabled by default and can only be turned off via the :ref:`JSON-input-output interface <compiler-api>`.

Reviewer 2: We usually refer to it as Standard JSON:
```suggestion
    enabled by default and can only be turned off via the :ref:`Standard JSON <compiler-api>`.
```

Author: Ok. Done.
